src/runtime_src/xrt/scheduler/sws.cpp

```diff
@@ -769,6 +769,11 @@ class xocl_scheduler
   {
     wait();
     queue_cmds();
+
+    // throttle polling for cu completion
+    if (auto us = xrt::config::get_polling_throttle())
```

Comment:
This throttles not only polling but also starting of CUs.
Move this code into the wait() function.
Think about changing it to be more intelligent, e.g. only sleep if no new commands have been added or no running commands have completed since last iteration.

Reply:
Thanks for the comment. I will provide another commit in this PR.